src/libtiled/wangset.cpp

```diff
@@ -0,0 +1,304 @@
+/*
+ * wangset.cpp
+ * Copyright 2017, Benjamin Trotter <bdtrotte@ucsc.edu>
+ * This file is part of libtiled.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ *    1. Redistributions of source code must retain the above copyright notice,
+ *       this list of conditions and the following disclaimer.
+ *
+ *    2. Redistributions in binary form must reproduce the above copyright
+ *       notice, this list of conditions and the following disclaimer in the
+ *       documentation and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE CONTRIBUTORS ``AS IS'' AND ANY EXPRESS OR
+ * IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF
+ * MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO
+ * EVENT SHALL THE CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+ * PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS;
+ * OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY,
+ * WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR
+ * OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF
+ * ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include "tilelayer.h"
+#include "wangset.h"
+
+#include <QStack>
+
+using namespace Tiled;
+
+unsigned cellToTileInfo(const Cell &cell)
+{
+    return cell.tileId()
+            | (cell.flippedHorizontally() << 29)
+            | (cell.flippedVertically() << 28)
+            | (cell.flippedAntiDiagonally() << 27);
+}
+
+unsigned wangTileToTileInfo(const WangTile &wangTile)
+{
+    return wangTile.tile()->id()
+            | (wangTile.flippedHorizontally() << 29)
+            | (wangTile.flippedVertically() << 28)
+            | (wangTile.flippedAntiDiagonally() << 27);
+}
+
+void WangTile::translate(int map[])
+{
+    int mask = (mFlippedHorizontally << 2)
+            | (mFlippedVertically << 1)
+            | (mFlippedAntiDiagonally);
+
+    mask = map[mask];
+
+    mFlippedHorizontally = mask & 4;
+    mFlippedVertically = mask & 2;
+    mFlippedAntiDiagonally = mask & 1;
+}
+
+void WangTile::rotateRight()
+{
+    int map[] = {5, 4, 1, 0, 7, 6, 3, 2};
+
+    translate(map);
+}
+
+void WangTile::rotateLeft()
+{
+    int map[] = {3, 2, 7, 6, 1, 0, 5, 4};
+
+    translate(map);
+}
+
+void WangTile::flipHorizontally()
+{
+    int map[] = {4, 3, 6, 1, 0, 7, 2, 5};
+
+    translate(map);
+}
+
+void WangTile::flipVertically()
+{
+    int map[] = {2, 5, 0, 7, 6, 1, 4, 3};
+
+    translate(map);
+}
+
+Cell WangTile::makeCell() const
+{
+    if(!mTile)
```

Comment:
Please make coding style adjustments in all needed places, not just where I point them out. This is still one of 10 places where there is no space after the `if`.